crates/daphne-server/src/storage_proxy_connection/kv/mod.rs

```diff
@@ -252,28 +253,64 @@ impl<'h> Kv<'h> {
     }
 
     #[tracing::instrument(
-        name = "kv_put",
+        name = "kv_delete",
         skip_all,
         fields(key, prefix = std::any::type_name::<P>()),
     )]
-    pub async fn put<P>(&self, key: &P::Key, value: P::Value) -> Result<(), Error>
+    pub async fn put_internal<P>(
+        &self,
+        key: &P::Key,
+        value: P::Value,
+        expiration: Option<Time>,
+    ) -> Result<(), Error>
     where
         P: KvPrefix,
         P::Key: std::fmt::Debug,
+        P::Value: Serialize,
     {
         let key = Self::to_key::<P>(key);
+        // let expiration_str = expiration.map_or_else(String::new, |time| time.to_string());
         tracing::debug!(key, "PUT");
-        self.http
+
+        let mut request = self
+            .http
             .post(self.config.url.join(&key).unwrap())
             .bearer_auth(&self.config.auth_token)
-            .body(serde_json::to_vec(&value).unwrap())
-            .send()
-            .await?
-            .error_for_status()?;
+            .body(serde_json::to_vec(&value).unwrap());
+
+        if let Some(expiration) = expiration {
+            request = request.header("X-Expiration", expiration.to_string());
+        }
+
+        request.send().await?.error_for_status()?;
+
         self.cache.write().await.put::<P>(key, Some(value.into()));
         Ok(())
     }
 
+    pub async fn put_with_expiration<P>(
+        &self,
+        key: &P::Key,
+        value: P::Value,
+        expiration: Option<Time>,
```

Comment:
nit: since the caller is already opting into using expiration, consider making it not optional and construction `Some(expiration)` within the function.
```suggestion:
        expiration: Time,
```

Reply:
Changed it for `put_with_expiration`